### PR TITLE
Support for serviceMonitor

### PR DESCRIPTION
**Pull request by mshivanna**
Signed-off-by: Manjunath Shivanna <leomanjucs@gmail.com>

### CODE REVIEW

File: charts/timescaledb-single/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    honorLabels: true
+    port: {{ .Values.serviceMonitor.portName }}
+    path: {{ .Values.serviceMonitor.path }}
+    {{- if .Values.serviceMonitor.metricRelabelings }}
+    metricRelabelings: {{ toYaml .Values.serviceMonitor.metricRelabelings | nindent 6 }}
+    {{- end }}
+  jobLabel: "{{ .Release.Name }}"
+  selector:
+    matchLabels:
+      app: {{ template "timescaledb.name" . }}
```
Comment (ErmakovDmitriy) [inline charts/timescaledb-single/templates/servicemonitor.yaml:34]: Hi, I am not the chart's maintainer but I tested your PR.

It seems that the Service which you try to target has an `app: {{ template "timescaledb.fullname" . }}` label.
I had to modify it to this value to monitor my deployment so my ServiceMonitor has:

```yaml
  selector:
    matchLabels:
      app: {{ template "timescaledb.fullname" . }}
      release: "{{ .Release.Name }}"
      component: postgres-exporter
```

**Comment by mshivanna** [comment]
@ErmakovDmitriy Would you mind review this and merge this ? Thank you

**Comment by ErmakovDmitriy** [comment]
Hi @mshivanna,

I am not a maintainer, I can make a review with "Approve" but I am not sure that it will work.
Have you taken a look at the ServiceMonitor match labels, in the "app" label's value?

File: charts/timescaledb-single/templates/servicemonitor.yaml (hunk 1 above)
Comment (mshivanna) [reply charts/timescaledb-single/templates/servicemonitor.yaml:34]: You are right. Adding the label ```component: postgres-exporter```  monitors the svc which has 3 endpoints(3 pods-1 master and 2 replicas). I will make changes to the PR accordingly. Thank you

Comment (paulfantom) [reply charts/timescaledb-single/templates/servicemonitor.yaml:34]: @mshivanna this is the only thing preventing me from merging your PR :) Could you update it?

Comment (mshivanna) [reply charts/timescaledb-single/templates/servicemonitor.yaml]: @paulfantom Have fixed it. sorry about the late fix. Changed selector match label from app: {{ template "timescaledb.name" . }} to app: {{ template "timescaledb.fullname" . }}

**Review by paulfantom — APPROVED** [review]
lgtm

**Pull request merged**